src/Private/Helpers.ps1

```diff
@@ -1766,22 +1766,21 @@ function Test-PodePath {
 
     }
 
-    # if we failed to get the file, report back the status code and/or return true/false
-    if (!$NoStatus.IsPresent) {
-        Set-PodeResponseStatus -Code $statusCode
+    if ($statusCode -eq 200) {
+        if ($ReturnItem) {
+            return  $item
+        }
+        return $true
     }
 
     if ($ReturnItem) {
-        if ($statusCode -eq 200) {
-            return  $item, $statusCode
-        }
-        else {
-            return  $null, $statusCode
+        # if we failed to get the file, report back the status code and/or return true/false
+        if (!$NoStatus.IsPresent) {
+            Set-PodeResponseStatus -Code $statusCode
         }
```

Comment:
Lines 1777-1780, for setting the status code, need to be moved up to line 1775 above the `if ($ReturnItem)` check, otherwise we're not setting the status code when needed. `$NoStatus` and `$ReturnItem` should be independent of one another.

```powershell
# if we failed to get the file, report back the status code and/or return true/false
if (!$NoStatus.IsPresent) {
    Set-PodeResponseStatus -Code $statusCode
}

if ($ReturnItem) {
    return  $null
}
return $false
```

For example, in `examples/web-pages-simple.ps1`, if you change the `Write-PodeViewResponse -Path 'simple'` to `Write-PodeViewResponse -Path 'simple2'` instead this will now return a 200 response and not a 404 like it does today.